ludwig/model_export/coreml_exporter.py

```diff
@@ -0,0 +1,48 @@
+from abc import ABC, abstractmethod
+
+import coremltools as ct
+import torch
+
+from ludwig.api import LudwigModel
+from ludwig.model_export.base_model_exporter import BaseModelExporter
+
+
+class CoreMLExporter(BaseModelExporter):
+    def forward(self, x):
+        return self.model({"image_path": x})
+
+    def export_classifier(self, model_path, export_path):
+        ludwig_model = LudwigModel.load(model_path)
+        model = CoreMLExporter(ludwig_model.model)  # Wrap the model
+
+        width = ludwig_model.config["input_features"][0]["preprocessing"]["width"]
+        height = ludwig_model.config["input_features"][0]["preprocessing"]["height"]
+        example_input = torch.randn(1, 3, width, height, requires_grad=True)
+
+        # Create the model and load the weights
+        ludwig_model.load_state_dict(torch.load(model_path))
```

Comment:
I tried running this and am getting this error:
```
Traceback (most recent call last):
  File "/palapa/auto-ml/export_classifier_coreml.py", line 82, in <module>
    export_classifier_coreml(model_id_from_args)
  File "/palapa/auto-ml/export_classifier_coreml.py", line 77, in export_classifier_coreml
    model.export_classifier(path, coreml_path)
  File "/palapa/auto-ml/export_classifier_coreml.py", line 38, in export_classifier
    ludwig_model.load_state_dict(torch.load(model_path))
AttributeError: 'LudwigModel' object has no attribute 'load_state_dict'
```

Here is the code I'm running it with:
```
def export_classifier_coreml(model_id):
    path = f"/palapa/auto-ml/saved_models/{model_id}/run/model"
    coreml_path = f"/palapa/auto-ml/saved_models/{model_id}/weights/best.mlmodel"

    ludwig_model = LudwigModel.load(path)
    model = CoreMLExporter(ludwig_model.model)

    model.export_classifier(path, coreml_path)
```